reflectapi/src/codegen/mod.rs

```diff
@@ -25,3 +29,99 @@ fn tmp_path(src: &str) -> PathBuf {
 
     std::env::temp_dir().join(format!("reflectapi-{hash}"))
 }
+
+trait Subst {
+    fn subst(self, subst: &HashMap<String, TypeReference>) -> Self;
+}
+
+trait Instantiate {
+    fn instantiate(self, args: &[TypeReference]) -> Self;
+}
+
+impl Subst for TypeReference {
+    fn subst(self, subst: &HashMap<String, TypeReference>) -> Self {
+        match subst.get(&self.name) {
+            Some(ty) => {
+                assert!(
+                    self.arguments.is_empty(),
+                    "type parameter cannot have type arguments (no HKTs)"
```

Comment:
better not to use acronyms. Use full name

Reply:
Just removed the higher-kinded types reference, not particular useful.